deepchecks/utils/abstracts/train_test_samples_mix.py

```diff
@@ -0,0 +1,51 @@
+# ----------------------------------------------------------------------------
+# Copyright (C) 2021-2023 Deepchecks (https://www.deepchecks.com)
+#
+# This file is part of Deepchecks.
+# Deepchecks is distributed under the terms of the GNU Affero General
+# Public License (version 3 or later).
+# You should have received a copy of the GNU Affero General Public License
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+"""module contains base logic for the train-test samples mix checks."""
+import typing as t
+import abc
+
+from typing_extensions import Self
+
+from deepchecks.core import ConditionCategory, ConditionResult
+from deepchecks.utils.strings import format_percent
+
+__all__ = ['TrainTestSamplesMixAbstract']
+
+
+class TrainTestSamplesMixAbstract(abc.ABC):
```

Comment:
Also here see if you can add more common logic - specifically the display and value calculation based on some common format

Reply:
look at the [comment](https://github.com/deepchecks/deepchecks/pull/2471#discussion_r1178681658) above